src/components/UserProfile/index.jsx

```diff
@@ -101,8 +102,25 @@ class UserProfile extends React.Component {
     };
 
     const getMode = (name) => {
-      if (name === this.props.currentlyEditingField) return 'editing';
-      if (!this.props[name] || !this.props[name].length) return 'empty';
+      // If the prop doesn't exist, that means it hasn't been set (for the current user's profile)
+      // or is being hidden from us (for other users' profiles)
+      const propExists = this.props[name] != null && this.props[name].length > 0;
+      // If this isn't the current user's profile, then there are only two options:
+      // static or nothing. We use 'null' as a return value because the consumers of
+      // getMode render nothing at all on a mode of null.
+      if (!isCurrentUserProfile) {
```

Comment:
I'm pondering if viewingOwnProfile or something similar would work?  isCurrentUserProfile makes me wonder if there are two profiles in play, one is the current one and one is the previous profile.  But it's also entirely possible that that's just me.  :-)

Reply:
Ah - well, just depends on our language for who the active session is.  I've tended to always call that the "current user" or the "active user" or something like that, so that's "is the profile of the current user", rather than "is the current profile that we're viewing". :) If we have some guidelines there I'm more than happy to line up with em!